src/site/SearchController.php

```diff
@@ -0,0 +1,71 @@
+<?hh // strict
+
+use HHVM\UserDocumentation\GuidesIndex;
+use HHVM\UserDocumentation\APIIndex;
+use HHVM\UserDocumentation\SearchResultSet;
+
+final class SearchController extends WebPageController {
+  private string $term;
+  private bool $term_set;
+
+  public function __construct(ImmMap<string, string> $params) {
+    parent::__construct($params);
+    $this->term = "";
+    $this->term_set = false;
+  }
+
+  protected async function getTitle(): Awaitable<string> {
+    return 'Search for shit!';
+  }
+
+  private function getListFromResultSet(Map<string, string> $result_set): XHPRoot {
+    $list = <ul />;
+    foreach ($result_set as $name => $path) {
+        $item = <li><a href={$path}>{$name}</a></li>;
+        $list->appendChild($item);
+    }
+    if (count($result_set) === 0) {
+        $item = <li><em>No results for this category</em></li>;
```

Comment:
Nit: bit weird to still use a `<ul><li>` for this - could some other markup be better?